platform/entanglement/src/capabilities/LocationCapability.js

```diff
@@ -0,0 +1,77 @@
+/*global define */
+
+define(
+    function () {
+        "use strict";
+
+        function LocationCapability(domainObject) {
+            this.domainObject = domainObject;
+            return this;
+        }
+
+        /**
+         * Persist the current location of the current domain object as it's
+         * primary location.  Returns a promise.
+         */
+        LocationCapability.prototype.persistLocation = function () {
```

Comment:
Worried the wording/naming is a little ambiguous here as well (I have to read the code to understand the consequences)

I'd either:
* Tweak the naming/wording for clarity.
* Move out the details to a `locate` action which does what this says when performed, and use that where this method is normally called. Not something to expose to the user, but if we ever decide that's useful, we could.